projects/core/utils/mask/create-number-mask.ts

```diff
@@ -75,7 +75,7 @@ export function tuiCreateNumberMask({
             : integer;
         const integerCappedClean = integerCapped
             .replace(TUI_NON_DIGITS_REGEXP, '')
-            .replace(/^0+(?!\.|$)/, '0');
+            .replace(/^0+(?!\.|$)/, '');
```

Comment:
add tests please for this case))

Reply:
Struggling to understand what this line does :D

Reply:
@Mahakon can you clarify what this line was doing and what is the new behavior? It's been a while since I looked into this code.

Reply:
so here is leaved one zero as a flag to add NON_ZERO_DIGIT prevent zeroes

Reply:
because of this extra symbol it do not write separate number with ThousandsSeparator